velPaint

```diff
@@ -138,16 +139,16 @@ while (my $line = <>) {
 					}
 				}
 				if ($lastSegOut != $nSegs) {
-					&outMove($x, $y, $z, $e);
+					outMove($x, $y, $z, $e);
 				}
 			}
 		}
-		$oldX = $x; $oldY = $y; $oldZ = $z; $oldE = $e;
+		($oldX, $oldY, $oldZ, $oldE) = ($x, $y, $z, $e);
 	} else {
-		if ($line =~ /G1 X([^\s]+) Y([^\s]+) Z([^\s]+)/) {
-			$oldX = $1; $oldY=$2; $oldZ = $3;
-		} elsif ($line =~ /G1 X([^\s]+) Y([^\s]+)/) {
-			$oldX = $1; $oldY=$2;
+		if ($line =~ /G1 X(\S+) Y(\S+) Z(\S+)/) {
+			($oldX, $oldY, $oldZ) = ($1, $2, $3);
```

Comment:
Would have used the same technique with %+ as above, except we don't want to overwrite `$oldZ` here. The code to avoid that would reduce the improvement in this code.